api/restapi/configure_keptn.go

```diff
@@ -88,7 +90,7 @@ func configureAPI(api *operations.KeptnAPI) http.Handler {
 
 	//api.EvaluationTriggerEvaluationHandler = evaluation.TriggerEvaluationHandlerFunc(handlers.TriggerEvaluationHandlerFunc)
 
-	rateLimiter := custommiddleware.NewRateLimiter(env.AuthRequestsPerSecond, env.AuthRequestMaxBurst, tokenValidator, clock.New())
+	rateLimiter := custommiddleware.NewRateLimiter(env.AuthEnabled, env.AuthRequestsPerSecond, env.AuthRequestMaxBurst, tokenValidator, clock.New())
```

Comment:
i think it would be easier to not call `api.AddMiddlewareFor(http.MethodPost, "/auth", rateLimiter.Handle)` if `env.AuthEnabled` is se to false